src/toasts/toast.component.js

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { always } from "kremling";
+
+const defaultOptions = {
+  millis: 4000
+};
+
+export default function Toast({ toast, closeToastRef, isClosing }) {
+  const { millis, description } = Object.assign({}, toast, defaultOptions);
+
+  const [waitingForTime, setWaitingForTime] = React.useState(true);
+  const [isMounting, setIsMounting] = React.useState(true);
+
+  React.useEffect(() => {
+    if (waitingForTime) {
+      const timeoutId = setTimeout(() => {
+        closeToastRef.current(toast);
```

Comment:
See the "setTimeout gotchas" section of [this article](https://upmostly.com/tutorials/settimeout-in-react-components-using-hooks) to see why we're using a ref here instead of just a normal callback function.